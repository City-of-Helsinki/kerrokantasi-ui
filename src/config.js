import { pluginMap } from './shared_config.json';

const config = {
  pluginMap,
  languages: ['fi', 'sv', 'en'],
  activeLanguage: 'fi',
  publicUrl: (typeof window !== 'undefined' ? window.PUBLIC_URL : null) || 'http://localhost:8086/',
  apiBaseUrl: (typeof window !== 'undefined' ? window.API_BASE_URL : null) || 'http://localhost:8000/',
  uiConfig: typeof window !== 'undefined' ? window.UI_CONFIG : null,
  openIdClientId: typeof window !== 'undefined' ? window.OPENID_CLIENT_ID : null,
  openIdAudience: typeof window !== 'undefined' ? window.OPENID_AUDIENCE : null,
  openIdAuthority: typeof window !== 'undefined' ? window.OPENID_AUTHORITY : null,
  openIdApiTokenUrl: typeof window !== 'undefined' ? window.OPENID_APITOKEN_URL : null,
  heroImageURL: (typeof window !== 'undefined' ? window.HERO_IMAGE_URL : null)
    || 'http://materialbank.myhelsinki.fi/detail/1192/download/7',
  showAccessibilityInfo: typeof window !== 'undefined' ? window.SHOW_ACCESSIBILITY_INFO : false,
  showSocialMediaSharing: typeof window !== 'undefined' ? window.SHOW_SOCIAL_MEDIA_SHARING : true,
  enableHighContrast: typeof window !== 'undefined' ? window.ENABLE_HIGHCONTRAST : false,
  enableStrongAuth: typeof window !== 'undefined' ? window.ENABLE_STRONG_AUTH : false,
  showCookiebar: typeof window !== 'undefined' ? window.SHOW_COOKIEBAR : false,
  adminHelpUrl: typeof window !== 'undefined' ? window.ADMIN_HELP_URL : "",
};

export default config;
