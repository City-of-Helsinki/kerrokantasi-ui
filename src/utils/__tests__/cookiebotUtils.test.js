/* eslint-disable import/no-unresolved */
import React from 'react';
import { render } from '@testing-library/react';

import {
  cookieBotAddListener,
  cookieBotRemoveListener,
  cookieBotImageOverride,
  getCookieBotConsentScripts,
  getCookieBotScripts,
  isCookiebotEnabled,
} from '../cookiebotUtils';
import config from '../../config';


jest.mock('../../config', () => ({
  enableCookies: true,
  enableCookiebot: true,
  cookiebotDataCbid: '123-abc'
}));

describe('cookiebotUtils', () => {
  describe('cookieBotAddListener', () => {
    afterEach(() => {
      jest.clearAllMocks();
      config.enableCookies = true;
      config.enableCookiebot = true;
    });
    it('calls window.addEventListener with correct params if enableCookies and enableCookiebot is true', () => {
      window.addEventListener = jest.fn();
      cookieBotAddListener();
      expect(window.addEventListener).toHaveBeenCalledWith('CookiebotOnDialogDisplay', cookieBotImageOverride);
    });
    it('does not call window.addEventListener when enableCookies is false', () => {
      config.enableCookies = false;
      window.addEventListener = jest.fn();
      cookieBotAddListener();
      expect(window.addEventListener).not.toHaveBeenCalled();
    });
    it('does not call window.addEventListener when enableCookiebot is false', () => {
      config.enableCookiebot = false;
      window.addEventListener = jest.fn();
      cookieBotAddListener();
      expect(window.addEventListener).not.toHaveBeenCalled();
    });
  });

  describe('cookieBotRemoveListener', () => {
    afterEach(() => {
      jest.clearAllMocks();
      config.enableCookies = true;
    });
    it('calls window.removeEventListener with correct params if enableCookies and enableCookiebot is true', () => {
      window.removeEventListener = jest.fn();
      cookieBotRemoveListener();
      expect(window.removeEventListener).toHaveBeenCalledWith('CookiebotOnDialogDisplay', cookieBotImageOverride);
    });
    it('does not call window.removeEventListener when enableCookies is false', () => {
      config.enableCookies = false;
      window.removeEventListener = jest.fn();
      cookieBotRemoveListener();
      expect(window.removeEventListener).not.toHaveBeenCalled();
    });
    it('does not call window.removeEventListener when enableCookiebot is false', () => {
      config.enableCookiebot = false;
      window.removeEventListener = jest.fn();
      cookieBotRemoveListener();
      expect(window.removeEventListener).not.toHaveBeenCalled();
    });
  });

  describe('getConsentScripts', () => {
    it('returns the Cookiebot script element', () => {
      const element = getCookieBotConsentScripts();

      const { container } = render(<div>{element}</div>);

      expect(container).toMatchSnapshot();
    });
  });
  describe('getCookieScripts', () => {
    it('returns a script element', () => {
      const element = getCookieBotScripts();

      const { container } = render(<div>{element}</div>);

      expect(container).toMatchSnapshot();
    });
  });

  describe('isCookiebotEnabled', () => {
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('returns true when both cookies and Cookiebot are enabled', () => {
      config.enableCookies = true;
      config.enableCookiebot = true;
      expect(isCookiebotEnabled()).toBe(true);
    });

    it('returns false when cookies are not enabled', () => {
      config.enableCookies = false;
      config.enableCookiebot = true;
      expect(isCookiebotEnabled()).toBe(false);
    });

    it('returns false when Cookiebot is not enabled', () => {
      config.enableCookies = true;
      config.enableCookiebot = false;
      expect(isCookiebotEnabled()).toBe(false);
    });
  });
});
