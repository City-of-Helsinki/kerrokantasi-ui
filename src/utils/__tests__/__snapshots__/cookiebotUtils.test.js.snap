// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cookiebotUtils getConsentScripts returns the Cookiebot script element 1`] = `
<div>
  <div>
    <script
      data-blockingmode="auto"
      data-cbid="123-abc"
      id="Cookiebot"
      src="https://consent.cookiebot.com/uc.js"
      type="text/javascript"
    />
  </div>
</div>
`;

exports[`cookiebotUtils getCookieScripts returns a script element 1`] = `
<div>
  <div>
    <script
      data-cookieconsent="statistics"
      type="text/plain"
    />
  </div>
</div>
`;
